### PR TITLE
add a Rust client for Pageserver page_service

### PR DESCRIPTION
**Pull request by problame**
Part of getpage@lsn benchmark epic: https://github.com/neondatabase/neon/issues/5771

Stacked atop https://github.com/neondatabase/neon/pull/6145

### CODE REVIEW

File: libs/pageserver_api/src/models.rs [hunk 1]
```diff
@@ -768,6 +769,59 @@ impl PagestreamBeMessage {
 
         bytes.into()
     }
+
+    pub fn deserialize(buf: Bytes) -> anyhow::Result<Self> {
+        let mut buf = buf.reader();
+        let msg_tag = buf.read_u8()?;
+        let ok = match msg_tag {
+            100 => {
```
[the remark this answers is not in the document]
Comment (problame) [reply libs/pageserver_api/src/models.rs:777]: I think the value of getting this PR merged without waiting for the CI is greater.

Remember that the tags are already duplicated in the C code base.

File: pageserver/client/src/page_service.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        };
+        let req = PagestreamFeMessage::GetPage(req);
+        let req: bytes::Bytes = req.serialize();
+        // let mut req = tokio_util::io::ReaderStream::new(&req);
+        let mut req = tokio_stream::once(Ok(req));
+
+        self.copy_both.send_all(&mut req).await?;
+
+        let next: Option<Result<bytes::Bytes, _>> = self.copy_both.next().await;
+        let next = next.unwrap().unwrap();
```
[the remark this answers is not in the document]
Comment (problame) [reply pageserver/client/src/page_service.rs:135]: Sigh, correct, should bail instead.

I guess I can [fix the tags](https://github.com/neondatabase/neon/pull/6128#discussion_r1430356368) then as well.

File: libs/pageserver_api/src/models.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (problame) [reply libs/pageserver_api/src/models.rs]: d4c9a0bbeb19c075b247448a3fb71b864111d4bc

File: pageserver/client/src/page_service.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (problame) [reply pageserver/client/src/page_service.rs]: d4c9a0bbeb19c075b247448a3fb71b864111d4bc

**Review by jcsp — APPROVED** [review]

**Pull request merged**